includes/class-wc-payments-webhook-processing-service.php

```diff
@@ -357,10 +357,13 @@ private function process_webhook_payment_intent_amount_capturable_updated( $even
 	 */
 	private function process_webhook_payment_intent_failed( $event_body ) {
 		// Check to make sure we should process this according to the payment method.
-		$charges_data        = $event_body['data']['object']['charges']['data'][0] ?? null;
-		$payment_method_type = $charges_data['payment_method_details']['type'] ?? null;
+		$charge_id           = $event_body['data']['object']['charges']['data'][0]['id'] ?? '';
+		$last_payment_error  = $event_body['data']['object']['last_payment_error'] ?? null;
+		$payment_method      = $last_payment_error['payment_method'] ?? null;
+		$payment_method_type = $payment_method['type'] ?? null;
```

Comment:
I see that originally the value of `$payment_method_type` is from `data.object.charges.data[0].payment_method_details.type` and now it's from `data.object.last_payment_error.type`.

I don't understand why that change. Will we always get `data.object.last_payment_error` for `payment_intent.payment_failed` webhook event despite the failing payment method (`card`, `us_bank_account`, or `au_becs_debit`)?

Will this change work when payment intent failed for `us_bank_account` and `au_becs_debit`? I don't know how to test that. Do you know?

Reply:
Reply to this and:
> I see that originally the value of $payment_method_id is from data.object.charges.data[0].payment_method and now it's from data.object.last_payment_error.id.

I mentioned it on the PR description, but I'll give more context here 🙂 

There are some `payment_intent.payment_failed` that doesn't have any charges. With our current implementation that will result in an on hold order with no additional notes. Instead, we can rely on `data.object.last_payment_error` which contains the error reason, among other things.

There is no existing issue, but we've seen similar cases – a recent Slack conversation about it p1666137895802399/1665973745.206889-slack-CGGCLBN58

It will also work for `us_bank_account` and `au_becs_debit`. As an example [here is a BECS failed intent](https://dashboard.stripe.com/test/connect/accounts/acct_1M4juUQoftj2Jyyc/payments/pi_3M4jyVQoftj2Jyyc17kpQBwj).

To test BECS you'll need to onboard an AU Stripe account, and manually request BECS capability through Stripe's dashboard before being able to enable it in WCPay Settings.

Reply:
Thank you for explaining all that. Good to know that `data.object.last_payment_error` will always be available and it's a better alternative than reading from charges.